src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -1352,31 +1609,46 @@ internal ProcessInputWriter(InternalCommand command)
         {
             Dbg.Assert(command != null, "Caller should validate the parameter");
             _command = command;
+
+            _pipeline = ScriptBlock.Create("Out-String -Stream").GetSteppablePipeline();
+            _pipeline.Begin(true);
         }
```

Comment:
The existing approach was spinning a thread and managing its lifecycle. It's pretty wasteful and unnecessary complicated. One of the my goal is to avoid a "medium" between two native processes, while implementing #1908. It would be a simple redirection without marshaling. So this would be a completely different code path. If we will try to reuse existing approach with `ProcessInputWriter` and enhance it for bytes, powershell would never be able to get closer to `bash` in `native | native` case by it's performance. We need to strip away the overhead of marshaling objects and just do the stream redirection.